sympy/vector/coordsysrect.py

```diff
@@ -438,6 +438,12 @@ def orient_new(self, name, orienters, location=None,
                 final_matrix = orienters.rotation_matrix(self)
             else:
                 final_matrix = orienters.rotation_matrix()
+            # TODO: trigsimp is needed here so that the matrix becomes
+            # canonical (scalar_map also calls trigsimp; without this, you can
+            # end up with the same CoordinateSystem that compares differently
+            # due to a differently formatted matrix). However, this is
+            # probably not so good for performance.
+            final_matrix = trigsimp(final_matrix)
```

Comment:
Yeah, there is a design issue to be worked out here, and I'm not the one to do it. The issue is that internal matrices are being rewritten (using trigsimp, i.e., in scalar_map), but then used for equality comparison. You'll need to think about how you want to do canonicalization and/or equality testing.